### PR TITLE
Adding packages for replacement.

### PR DESCRIPTION
**Pull request by jgallimore**
Further package rename rules - I'm not sure the current set is complete. This is the set of rules we currently use on TomEE.

### CODE REVIEW

**Comment by jgallimore** [comment]
I'm not aware of an official one, I'm afraid. Here's the list I've been working with. https://gist.github.com/dblevins/9a6d4b1c90986a4116dd738c9e5ef212

**Comment by dblevins** [comment]
@bjhargrave we're looking to get that gist on the jakarta.ee site somewhere at some point.

**Comment by bjhargrave** [comment]
>  Here's the list I've been working with.

That's great. Ultimately we will need a broader list which includes things beyond packages. I know xml schemas are being renamed, properties are being renamed, etc. 

Jakarta renaming is the gift that keeps on giving...

**Comment by bjhargrave** [comment]
@tbitonti Can you please review these updates so we can merge? Thanks.

**Comment by dblevins** [comment]
> Jakarta renaming is the gift that keeps on giving...

LOL, so true.  And when it comes to JSF it's like Christmas. :)

They've got their namespace in Javascript files, TLD files, html files, ....  I took shot at some renaming on the TomEE side, but still not sure if it worked.

**Comment by tbitonti** [comment]
Reviewing ...

**Comment by bjhargrave** [comment]
https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/platform/jakarta.jakartaee-api/9.0.0-RC2/jakarta.jakartaee-api-9.0.0-RC2.jar seems to be the very latest complete list of Jakarta EE 9 APIs. We should review the packages in this jar against the package list being updated in this API.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties [hunk 1]
```diff
@@ -25,42 +25,81 @@
 javax.activation=jakarta.activation
 
 javax.annotation=jakarta.annotation
+javax.annotation.concurrent=jakarta.annotation.concurrent
```
Comment (bjhargrave) [inline org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:28]: This package does not seem to be part of the Jakarta EE 9 RC2 api.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties [hunk 2]
```diff
@@ -25,42 +25,81 @@
...
+javax.batch.api.chunk=jakarta.batch.api.chunk
+javax.batch.api.chunk.listener=jakarta.batch.api.chunk.listener
+javax.batch.api.listener=jakarta.batch.api.listener
+javax.batch.api.partition=jakarta.batch.api.partition
+javax.batch.operations=jakarta.batch.operations
+javax.batch.runtime=jakarta.batch.runtime
+javax.batch.runtime.context=jakarta.batch.runtime.context
 
-javax.el=jakarta.el
 com.ibm.wsspi.el=com.ibm.wsspi.el
```
Comment (bjhargrave) [inline org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:41]: This line  should be removed. It is some IBM leftovers...

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties [hunk 3]
```diff
@@ -72,35 +111,65 @@ javax.resource.spi=jakarta.resource.spi
...
-javax.servlet.jsp.resources=jakarta.servlet.jsp.resources
-javax.servlet.jsp.tagext=jakarta.servlet.jsp.tagext
-
+javax.servlet.jsp.jstl=jakarta.servlet.jsp.jstl
 javax.servlet.jsp.jstl.core=jakarta.servlet.jsp.jstl.core
 javax.servlet.jsp.jstl.fmt=jakarta.servlet.jsp.jstl.fmt
 javax.servlet.jsp.jstl.sql=jakarta.servlet.jsp.jstl.sql
 javax.servlet.jsp.jstl.tlv=jakarta.servlet.jsp.jstl.tlv
-
+javax.servlet.jsp.resources=jakarta.servlet.jsp.resources
```
Comment (bjhargrave) [inline org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:136]: This package does not seem to be part of the Jakarta EE 9 RC2 api.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 2 above)
Comment (jgallimore) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:41]: Wasn't sure if you wanted to keep that or not :)

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 3 above)
Comment (jgallimore) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:136]: Is it possible to keep this? Tomcat's servlet api jar does include this package.

**Comment by jgallimore** [comment]
retest this please

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 3 above)
Comment (bjhargrave) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:133]: More googling seems to point out this package is legit. Not sure why it is not in the RC2 api jar.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 2 above)
Comment (bjhargrave) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties]: Some test cares about this IBM package. So put it back for now. We can remove it and change the test case in another PR.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 3 above)
Comment (jgallimore) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:135]: I'll take a look and see if I can spot anything missing, or a specific reason its included in the Apache Tomcat version of servlet-api.

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 2 above)
Comment (jgallimore) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:40]: Whoops! I've put that back.

**Comment by kwsutter** [comment]
> I think it is ready to merge.

Don't you want to wait until this jakarta.servlet.jsp.resources question is answered?

File: org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties (hunk 3 above)
Comment (kwsutter) [reply org.eclipse.transformer.cli/src/main/resources/org/eclipse/transformer/jakarta/jakarta-renames.properties:135]: This "resources" package is not part of the JSP API source:
https://github.com/eclipse-ee4j/jsp-api/tree/master/api/src/main/java/jakarta/servlet/jsp

At this point, I don't think this package should be included in the Transformer rules.

**Comment by kwsutter** [comment]
> Here's a link to the active discussion: https://www.eclipse.org/lists/jsp-dev/msg00092.html

This discussion just continues to evolve.  It's now encompassing a similar issue with missing dtd/xsd files for Servlet.  Since the API jar files for both Servlet and JSP do not contain these "resources" packages, I would recommend leaving them out of the Transformer official rule set.  Maybe the rules are specific to an implementation for now (ie. Tomcat).  Once we get these "resources" packages figured out, then we can revisit the rules for the Transformer.  That's how I would approach this...

**Comment by jgallimore** [comment]
The conversation sounds extremely useful, and I'm grateful to you, Mark and others for your thoughts.

My personal view is that it should be left in, as it sounds like it was removed in error at some point and the view from both Tomcat and Jetty is that it should be re-instated.

That said, later today, I'll try a run with TomEE transformed without this one rule, and see what the impact is. It may well be that the inclusion of this rule is probably a bigger deal for transforming servers than it is for applications.

**Comment by bjhargrave** [comment]
I think we leave it in for now as it helps a current user. 

We are still early days in this project and there will be many more times we will need to fine tune the Jakarta rules. I suspect we may need to add another resources package for servlet when the dust settles.

See, this project has already made a small contribution to improving the Jakarta EE 9 specs!

**Pull request merged**